indexer/ftypes_matcher.cpp

```diff
@@ -712,20 +712,112 @@ class RussiaRoadShieldParser : public RoadShieldParser
 
     strings::UniString s = strings::MakeUniString(rawText);
     if (s[0] == 'E' || s[0] == strings::UniChar(1045)) // Latin and cyrillic.
-      return RoadShield(RoadShieldType::Euro_Motorway, rawText);
+      return RoadShield(RoadShieldType::Generic_Green, rawText);
 
-    return RoadShield(RoadShieldType::Russia_Highway, rawText);
+    return RoadShield(RoadShieldType::Generic_Blue, rawText);
   }
 };
 
+class SimpleUnicodeRoadShieldParser : public RoadShieldParser
+{
+public:
+  struct Name
+  {
+    Name(char simpleName, strings::UniChar const & unicodeName)
+      : m_simpleName(simpleName), m_unicodeName(unicodeName)
+    {
+    }
+    char m_simpleName;
+    strings::UniChar m_unicodeName;
+  };
+
+  using ShieldTypes = std::vector<std::pair<Name, RoadShieldType>>;
+
+  SimpleUnicodeRoadShieldParser(std::string const & baseRoadNumber, ShieldTypes const & types)
+    : RoadShieldParser(baseRoadNumber), m_types(types)
+  {
+  }
+
+  RoadShield ParseRoadShield(std::string const & rawText) override
+  {
+    if (rawText.size() > kMaxRoadShieldTextSize)
+      return RoadShield();
+
+    for (auto const & p : m_types)
+    {
+      Name const & name = p.first;
+      RoadShieldType const type = p.second;
+
+      if (rawText.find(name.m_simpleName) != std::string::npos)
+        return RoadShield(type, rawText);
+
+      strings::UniString s = strings::MakeUniString(rawText);
```

Comment:
Do we really need a test for this case? There is check for an empty string above?